src/main/java/com/aws/greengrass/componentmanager/ClientConfigurationUtils.java

```diff
@@ -110,23 +120,37 @@ private static void configureClientMutualTLS(ApacheHttpClient.Builder httpBuilde
         }
     }
 
-    private static KeyManager[] createKeyManagers(String privateKeyPath, String certificatePath)
-            throws TLSAuthException {
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.PreserveStackTrace"})
+    KeyManager[] createKeyManagers(String privateKeyPath, String certificatePath) throws TLSAuthException {
         try {
-            List<X509Certificate> certificateChain = EncryptionUtils.loadX509Certificates(Paths.get(certificatePath));
-
-            PrivateKey privateKey = EncryptionUtils.loadPrivateKey(Paths.get(privateKeyPath));
-
-            KeyStore keyStore = KeyStore.getInstance("PKCS12");
-            keyStore.load(null);
-            keyStore.setKeyEntry("private-key", privateKey, null, certificateChain.toArray(new Certificate[0]));
+            return RetryUtils.runWithRetry(SECURITY_SERVICE_RETRY_CONFIG, () -> securityService
```

Comment:
Yeah, instead of trying to strictly sort services in order. The client could retry on the operations if service is not available. (BTW, the default file based implementation won't throw unavailable exception because it's guaranteed available).  
Infinite retry is because get SSL key manager is critical for the component deployment process. If it's not available, there is no meaning of moving forward. Customer needs to deploy proper key provider service to continue process if it's meant to use the service.

Reply:
This is what I am thinking, let me know if it makes sense. The problem with infinite retry is the case when the security service never comes up due to x reason. Then we just occupy the thread waiting for service and potentially brick greengrass, because no other code can execute in the meanwhile? If it gives up after x retries, then we can atleast inform the user by logging that "Startup failed because PKCS is not available". That way customers can try something different again. x could be a very large value, which kind of mimics infinite retry case, but still gives up and inform the user.

Reply:
It makes sense. I was not sure how long is 'long' enough, so started with 'infinite'. In valid case, if the client probes the service in unavailable state, it will only be 'unavailable' for a short while before it initialize properly. So a reasonable retry period should be fine. Let me change to finite retry.